apps/lottery/src/components/layout/header/account/account-button/AccountButton.tsx

```diff
@@ -0,0 +1,29 @@
+import Identicon from '@polkadot/react-identicon';
+import clsx from 'clsx';
+import { buttonStyles } from '@gear-js/ui';
+
+type Props = {
+  address: string;
+  name: string | undefined;
+  onClick: () => void;
+  isActive?: boolean;
+  block?: boolean;
+};
+
+function AccountButton({ address, name, onClick, isActive, block }: Props) {
+  const className = clsx(
+    buttonStyles.button,
+    buttonStyles.normal,
+    isActive ? buttonStyles.primary : buttonStyles.secondary,
+    block && buttonStyles.block,
+  );
+
+  return (
+    <button type="button" className={className} onClick={onClick}>
+      <Identicon value={address} className={buttonStyles.icon} theme="polkadot" size={28} />
+      {name}
```

Comment:
And if name is undefined? Maybe better to add some logic for this case?

Reply:
Dot extension won't allow you to create an account without name, so I don't sure why it can be `undefined` first place.